### PR TITLE
enable auto revendoring of mcm

### PR DESCRIPTION
**Pull request by kon-angelo**
**What this PR does / why we need it**:

Enable automatic revendoring of MCM

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement user
NONE
```

### CODE REVIEW

File: .ci/set_dependency_version [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+echo -n "${DEPENDENCY_VERSION}" > "${MCM_FILEPATH}"
+echo "set dependency-version of ${DEPENDENCY_NAME} to ${DEPENDENCY_VERSION}"
+
+cd ${MAIN_REPO_DIR}
+old_version=$(cat go.mod | grep "github.com/gardener/machine-controller-manager v" | xargs)
+new_version="github.com/gardener/machine-controller-manager${DEPENDENCY_VERSION}"
+
+sed -i -- 's#'"${old_version}"'#'"${new_version}"'#g' go.mod
+apk add --no-cache go make
+make revendor
```
Comment (himanshu-kun) [inline .ci/set_dependency_version:38]: maybe breaking it into two steps(`go mod tidy` and `go mod vendor`) could be better, as I saw last auto-revendoring in other providers failed using the make rule but worked by using these two commands seperately.

File: .ci/pipeline_definitions [hunk 2]
```diff
@@ -14,6 +14,13 @@ machine-controller-manager-provider-openstack:
             image: 'eu.gcr.io/gardener-project/gardener/machine-controller-manager-provider-openstack'
             target: machine-controller
   jobs:
+    create-upgrade-prs:
+      traits:
+        component_descriptor: ~
+        version: ~
+        cronjob:
+          interval: '5m'
```
Comment (himanshu-kun) [inline .ci/pipeline_definitions:22]: Vedran suggested to update this interval to `24h` , I did the change in one provider others are left though

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
two comments looks good otherwise, hope its running fine locally

File: .ci/set_dependency_version (hunk 1 above)
Comment (kon-angelo) [reply .ci/set_dependency_version]: I think the issue was that we were calling vendor before tidy. I changed that and seem to fix the problem in my tests

**Review by himanshu-kun — APPROVED** [review]
yes thanks , even I thought the same. 
/lgtm

**Pull request merged**